lib/pvoutput/client.rb

```diff
@@ -50,11 +53,66 @@ def add_output(options)
       params[:tm] = options[:min_temp] if options[:min_temp]
       params[:tx] = options[:max_temp] if options[:max_temp]
       params[:cm] = options[:comments] if options[:comments]
+      params[:ip] = options[:import_peak] if options[:import_peak]
+      params[:io] = options[:import_off_peak] if options[:import_off_peak]
+      params[:is] = options[:import_shoulder] if options[:import_shoulder]
+      params[:ih] = options[:import_high_shoulder] if options[:import_high_shoulder]
+      params[:c] = options[:consumption] if options[:consumption]
 
       response = self.class.post('/service/r2/addoutput.jsp', :body => params)
 
       raise('Bad Post') unless response.code == 200
     end
     # rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+    # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+    # rubocop:disable Style/Next
+    def add_batch_output(options)
+      params = {
+      }
+      data = ''
+      count = 0
+
+      options.each do |date, values|
+        data.concat "#{date},"
+        data.concat values[:energy_generated].to_s
```

Comment:
Maybe it would be neater to do something like

``` ruby
keys = %i(energy_generated energy_export energy_used)
keys += %i(peak_power peak_time) # etc to a sensible line length

data = keys.map { |key| values[key] }.join(',')
```